modules/swagger-parser/src/main/java/io/swagger/parser/SwaggerParser.java

```diff
@@ -103,7 +103,10 @@ public SwaggerDeserializationResult readWithInfo(String swaggerAsString) {
             if (result != null) {
                 result.setSwagger(new SwaggerResolver(result.getSwagger(), new ArrayList<AuthorizationValue>(), null).resolve());
             }
-            return new Swagger20Parser().readWithInfo(node);
+            else {
+                result.message("Definition does not appear to be a valid Swagger format");
+            }
+            return result;
```

Comment:
I have no idea why we were parsing, resolving, then re-parsing without re-resolving.  Seems like a bad joke, but in the end, we were returning an un-resolved model!